lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1668,43 +1669,33 @@ bool AsmPrinter::EmitSpecialLLVMGlobal(const GlobalVariable *GV) {
   report_fatal_error("unknown special variable");
 }
 
-void AsmPrinter::EmitPOT(const GlobalVariable *GV) {
-  // FIXME: This duplicates a lot of code from EmitGlobalVariable. Might be
-  // better to create the global variable earlier somehow and emit it normally.
-  MCSymbol *POTSym = OutContext.getOrCreateSymbol(StringRef("_POT_"));
-  EmitVisibility(POTSym, GV->getVisibility(), !GV->isDeclaration());
-
-  // uint64_t Size = DL->getTypeAllocSize(GV->getType()->getElementType());
-
-  unsigned AlignLog = getGVAlignmentLog2(GV, getDataLayout());
-  unsigned Align = 1 << AlignLog;
-
-  // for (const HandlerInfo &HI : Handlers) {
-  //   NamedRegionTimer T(HI.TimerName, HI.TimerGroupName, TimePassesIsEnabled);
-  //   HI.Handler->setSymbolSize(POTSym, Size);
-  // }
-
-  MCSection *TheSection =
-    getObjFileLowering().getSectionForConstant(getDataLayout(),
-                                               SectionKind::getReadOnlyWithRel(),
-                                               /*C=*/nullptr, Align);
-
-  OutStreamer->SwitchSection(TheSection);
+void AsmPrinter::EmitPOT() {
+  unsigned Alignment = getDataLayout().getPointerPrefAlignment(0);
+  unsigned PtrSize = getDataLayout().getPointerSize(0);
 
-  EmitLinkage(GV, POTSym);
-  EmitAlignment(AlignLog, GV);
+  auto *Section = getObjFileLowering().getSectionForConstant(
+      getDataLayout(), SectionKind::getReadOnlyWithRel(), nullptr, Alignment);
+  OutStreamer->SwitchSection(Section);
 
+  // Emit POT start label
+  MCSymbol *POTSym = OutContext.getOrCreateSymbol("_POT_");
```

Comment:
Good question. I was using external+protected mostly for ease in debugging. I think we should instead use internal linkage, which implies default visibility. That means we can omit both the `EmitSymbolAttribute` calls.